### ISSUE TITLE
Allow recorders to output enumerations as either their full value, or the integer

### ISSUE THREAD

**Issue by ftuffner** [gridlab-d/gridlab-d#1019]
Many objects have enumeration properties, but this can bloat CSV-type outputs significantly, and make parsing difficult.  The overall idea is to make it so a recorder has the ability to either write out the full enumeration keyword, or just the integer representation.

An idea for doing this could be something with the unit functionality of recorders.  Basically a:
`property enum_prop[enumeration]` or `property enum_prop[number]` type of thing.

**Comment by dchassin** [comment]
Good idea. Some comments/thoughts: The unit system assumes the underlying type of the property is double or complex, so that may be challenging to use for anything that has an underlying integer type.  One possibility is to have syntax like unit that effectively casts the underlying data to any desired type.  This would make the solution much more generally applicable to any property that normally gets converted on output.  The thinking would be to allow an arbitrary printf format that bypasses conversion and forces the data to be run through sprint instead.  Then we could have `enum_prop[%d]` to output as decimal int, or `enum_prop[%08x]` as 8 digit hex int with leading zeros, etc.  The only challenge is ensuring that the underlying type matches the format. For integers, enums, and sets this is easy.  Same for strings although the formatting can be a bit more complicated given the problems that can come up with sprintf on POD types. The tricky thing is making it work for underlying doubles because the [%...] syntax is already used by the existing unit system.  So we can't use % leading a string to flag special handling using sprintf instead of the unit system because % is a legit unit and can be part of a compound unit that could lead to ambiguous syntax.

**Comment by dchassin** [comment]
Another thought: in server mode we support an extended formatting string syntax that could be the basis for this approach.  The code is implemented in gldcore/server.c:get_value_with_unit(). It allows double values to not only have different unit, but also different number formatting.  See http://gridlab-d.sourceforge.net/wiki/index.php/Realtime_server#XML for details. Having units-like formatting for string and integer underlying types would be unambiguous if we used this same syntax but without the real-values units specification before the comma.

**Comment by dchassin** [comment]
The syntax for underlying strings would be `string_name[<num>s]` where <num> indicates the maximum string length.  For underlying integers it would be `dec_name[<num>d]`, `hex_name[<num>x]` or `oct_name[<num>o]`  where <num> is the maximum number of digits to output.  A leading zero in <num> would pad the output with leading zeros.
